R/diagnose_design.R

```diff
@@ -189,6 +193,29 @@ check_design_class <- function(designs){
 }
 
 calculate_diagnosands <- function(simulations_df, diagnosands, group_by_set) {
+  
+  if ("design_label" %in% group_by_set) {
+    group_by_list <- simulations_df[, "design_label", drop = FALSE]
+    labels_df <- split(group_by_list, group_by_list, drop = TRUE)
+    labels_df <- lapply(labels_df, head, n = 1)
+    
+    simulations_df <- split(simulations_df, group_by_list, drop = TRUE)
+    diagnosands_df <- lapply(seq_along(simulations_df), FUN = function(i) {  
+      dg <- calculate_diagnosands_single_design(
+        simulations_df[[i]], diagnosands[[i]], group_by_set[-1])
+      data.frame(labels_df[[i]], dg)
```

Comment:
you can pass the labels_df[[i]] to calculate_diagnosands_single_design also

Reply:
I think not, because labels_df in calculate_diagnosands is now just the designs -- doesn't include estimator_label etc. so I think needs to be recreated a level below (without design_label).